colossalai/accelerator/cuda_accelerator.py

```diff
@@ -29,6 +37,8 @@ def set_device(self, device: Union[torch.device, int]) -> None:
         """
         Bind the current process to a device.
         """
+        if device is None:
+            device = dist.get_rank() % self.device_count()
```

Comment:
If distributed is not initialized, `get_rank()` may raise error

Reply:
add an assert